File: selenium/test/java/com/thoughtworks/selenium/corebased/TestType.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.*;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import java.util.regex.Pattern;

public class TestType extends SeleneseTestNgHelper {
	@Test public void testType() throws Exception {
		selenium.open("../tests/html/test_type_page1.html");
		verifyEquals(selenium.getValue("username"), "");
		selenium.shiftKeyDown();
		selenium.type("username", "x");
		verifyEquals(selenium.getValue("username"), "X");
		selenium.shiftKeyUp();
		selenium.type("username", "TestUserWithLongName");
		verifyEquals(selenium.getValue("username"), "TestUserWi");
		selenium.type("username", "TestUser");
		verifyEquals(selenium.getValue("username"), "TestUser");
		verifyEquals(selenium.getValue("password"), "");
		selenium.type("password", "testUserPasswordIsVeryLong");
		verifyEquals(selenium.getValue("password"), "testUserPasswordIsVe");
		selenium.type("password", "testUserPassword");
		verifyEquals(selenium.getValue("password"), "testUserPassword");
		selenium.click("submitButton");
		selenium.waitForPageToLoad("30000");
		verifyTrue(selenium.isTextPresent("Welcome, TestUser!"));
	}
}

File: dotnet/test/webdriverbackedselenium/TestType.cs
using NUnit.Framework;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestType : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldTestTyping()
        {
            selenium.Open("/html/test_type_page1.html");
            Assert.AreEqual(string.Empty, selenium.GetValue("username"));
            selenium.ShiftKeyDown();
            selenium.Type("username", "x");
            Assert.AreEqual("X", selenium.GetValue("username"));
            selenium.ShiftKeyUp();
            selenium.Type("username", "TestUserWithLongName");
            Assert.AreEqual("TestUserWi", selenium.GetValue("username"));
            selenium.Type("username", "TestUser");
            Assert.AreEqual("TestUser", selenium.GetValue("username"));
            Assert.AreEqual(string.Empty, selenium.GetValue("password"));
            selenium.Type("password", "testUserPasswordIsVeryLong");
            Assert.AreEqual("testUserPasswordIsVe", selenium.GetValue("password"));
            selenium.Type("password", "testUserPassword");
            Assert.AreEqual("testUserPassword", selenium.GetValue("password"));
            selenium.Click("submitButton");
            selenium.WaitForPageToLoad("30000");
            Assert.IsTrue(selenium.IsTextPresent("Welcome, TestUser!"));
        }
    }
}
